balancer/grpclb/grpclb_test.go

```diff
@@ -832,11 +832,12 @@ func (s) TestGRPCLB_Fallback(t *testing.T) {
 		grpc.WithTransportCredentials(&serverNameCheckCreds{}),
 		grpc.WithContextDialer(fakeNameDialer),
 	}
-	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName, dopts...)
+	cc, err := grpc.NewClient(r.Scheme()+":///"+beServerName, dopts...)
 	if err != nil {
-		t.Fatalf("Failed to dial to the backend %v", err)
+		t.Fatalf("Failed to create new client to the backend %v", err)
 	}
 	defer cc.Close()
+	cc.Connect()
```

Comment:
This should be unnecessary, throughout.

If it is necessary, there should be a comment explaining exactly why.

I suspect the reason for this one is that we're updating the resolver state before the channel has started?  If it is needed here, you can use `r.InitialState` before creating the client instead of `r.UpdateState` afterwards.  But it's possible it works without the `Connect` call, even without that change.

Reply:
cc.Connect() to ensure that the resolver is built. The clientconn is passed to the resolver though the resolver.Build() method. grpc.NewClient is lazy, i.e. it doesn't start creating subchannels until an RPC is made or cc.Connect() is called.